sklearn/preprocessing/imputation.py

```diff
@@ -375,3 +375,131 @@ def transform(self, X):
             X[coordinates] = values
 
         return X
+
+
+class MissingIndicator(BaseEstimator, TransformerMixin):
+    """Missing values indicator transformer for indicating missing values.
+
+    Parameters
+    ----------
+    missing_values : integer or "NaN", optional (default="NaN")
+        The placeholder for the missing values. All occurrences of
+        `missing_values` will be imputed. For missing values encoded as np.nan,
+        use the string value "NaN".
+
+    features : [None, "all", "train", array(indices/mask)]
+        If None,
+        If "all", mask will represent all features
+        If "train", mask will only represent features with missing values
+        during fit time
+        If mask/indices, mask will only represent features in the
+        indices or mask
+
+    sparse : boolean or "auto", optional (default="auto")
+        If True, the transformed ``X`` will be sparse type.
+        If False, the transformed ``X`` will be dense type
+        If "auto", the transformed ``X`` will be os same type as input
+
+    Attributes
+    ----------
+    feat_with_missing_  : array of shape(n_missing_features, )
+            The features with missing values.
+    """
+
+    def __init__(self, missing_values="NaN", features=None, sparse="auto"):
+        self.missing_values = missing_values
+        self.features = features
+        self.sparse = sparse
+
+    def fit(self, X):
+        """Fit the transformer on X.
+        Parameters
+        ----------
+        X : {array-like, sparse matrix}, shape (n_samples, n_features)
+            Input data, where ``n_samples`` is the number of samples and
+            ``n_features`` is the number of features.
+        Returns
+        -------
+        self : object
+            Returns self.
+        """
+        if (isinstance(self.features, six.string_types) and
```

Comment:
Should probably also validate that:
* `features`, if an array-like, is an integer array
* `sparse` has a valid value